devolutions-gateway/src/config.rs

```diff
@@ -80,15 +80,39 @@ pub struct CertificateConfig {
     pub private_key_data: Option<String>,
 }
 
-#[derive(Debug, Default, Clone)]
-pub struct RecordingInfo {
-    pub sogar_path: Option<String>,
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+pub struct SogarPushRegistryInfo {
+    pub sogar_util_path: Option<String>,
     pub registry_url: Option<String>,
     pub username: Option<String>,
     pub password: Option<String>,
     pub image_name: Option<String>,
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum SogarPermission {
+    Push,
+    Pull,
+}
+
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+pub struct SogarUser {
+    pub password: Option<String>,
+    pub username: Option<String>,
+    pub permission: Option<SogarPermission>,
```

Comment:
A single user can't have both pull AND push permissions?

Reply:
When we were discussing the feature with @awakecoding there was a point that we don't need the "admin" user to manage the files. The users should have scoped permissions. That's why the user will only have the pull or push permission.

Reply:
Sounds good!